demos/python_demos/monitors_extension/monitors_extension.cpp

```diff
@@ -0,0 +1,228 @@
+int presenter_setGraphSize(PresenterObject *self, PyObject *value, void *closure) {
+    PyErr_Format(PyExc_RuntimeError, "Read-only attribute: graphSize");
+    return -1;
+}
+
+PyObject *presenter_getGraphPadding(PresenterObject *self, void *closure) {
+    if (!self->_presenter) {
+        PyErr_SetString(PyExc_AttributeError, "Underlying C++ presenter is nullptr");
+        return nullptr;
+    }
+    PyObject * graphPadding = PyLong_FromLong(self->_presenter->graphPadding);
+    if (PyErr_Occurred())
+        return nullptr;
+    return graphPadding;
+}
+
+int presenter_setGraphPadding(PresenterObject *self, PyObject *value, void *closure) {
+    PyErr_Format(PyExc_RuntimeError, "Read-only attribute: graphPadding");
+    return -1;
+}
+
+char graphPaddingName[] = "graphPadding";
+PyGetSetDef presenter_getsetters[] = {
+    {yPosName, (getter)presenter_getYPos, (setter)presenter_setYPos,},
+    {graphSizeName, (getter)presenter_getGraphSize, (setter)presenter_setGraphSize},
+    {graphPaddingName, (getter)presenter_getGraphPadding, (setter)presenter_setGraphPadding},
+    {nullptr}};  // Sentinel
+
+char monitors_extension_doc[] = "The module is a wrapper over C++ monitors. It guarantees that C++ and Python "
+    "monitors are consistent.";
+
+PyTypeObject presenterType{PyVarObject_HEAD_INIT(nullptr, 0)
+    "monitors_extension.Presenter",  // tp_name
+    sizeof(PresenterObject),  // tp_basicsize
+    0,  // tp_itemsize
+    reinterpret_cast<destructor>(presenter_dealloc),  // tp_dealloc
+    0,  // tp_vectorcall_offset
+    0,  // tp_getattr
+    0,  // tp_setattr
+    0,  // tp_as_async
+    0,  // tp_repr
+    0,  // tp_as_number
+    0,  // tp_as_sequence
+    0,  // tp_as_mapping
+    0,  // tp_hash
+    0,  // tp_call
+    0,  // tp_str
+    0,  // tp_getattro
+    0,  // tp_setattro
+    0,  // tp_as_buffer
+    0,  // tp_flags
+    monitors_extension_doc,  // tp_doc
+    0,  // tp_traverse
+    0,  // tp_clear
+    0,  // tp_richcompare
+    0,  // tp_weaklistoffset
+    0,  // tp_iter
+    0,  // tp_iternext
+    presneter_methods,  // tp_methods
+    0,  // tp_members
+    presenter_getsetters,  // tp_getset
+    0,  // tp_base
+    0,  // tp_dict
+    0,  // tp_descr_get
+    0,  // tp_descr_set
+    0,  // tp_dictoffset
+    reinterpret_cast<initproc>(presenter_init),  // tp_init
+    0,  // tp_alloc
+    PyType_GenericNew};  // tp_new
+
+PyModuleDef monitors_extension{PyModuleDef_HEAD_INIT, "monitors_extension", monitors_extension_doc, 0};
+}
+
+PyMODINIT_FUNC PyInit_monitors_extension() {
+    if (PyType_Ready(&presenterType) < 0) return nullptr;
+
+    PyObject *m = PyModule_Create(&monitors_extension);
+    if (m == nullptr) return nullptr;
+
+    Py_INCREF(&presenterType);
+    if (PyModule_AddObject(m, "Presenter", reinterpret_cast<PyObject *>(&presenterType)) < 0) {
+        Py_DECREF(&presenterType);
+        Py_DECREF(m);
+        return nullptr;
+    }
+
+    import_array();
+    if (PyErr_Occurred()) return nullptr;
```

Comment:
This leaks the module. I'd suggest to move the `import_array` call to before the module is created.

Reply:
Fixed